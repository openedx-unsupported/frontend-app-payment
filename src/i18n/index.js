import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
