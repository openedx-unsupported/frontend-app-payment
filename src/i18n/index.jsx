import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
