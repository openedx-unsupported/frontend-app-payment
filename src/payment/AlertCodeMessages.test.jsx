import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import {
  SingleEnrollmentCodeWarning,
  EnrollmentCodeQuantityUpdated,
  TransactionDeclined,
} from './AlertCodeMessages';

const mockStore = configureMockStore();

describe('SingleEnrollmentCodeWarning', () => {
  it('should render with values', () => {
    const component = (
      <IntlProvider locale="en">
        <SingleEnrollmentCodeWarning values={{ courseAboutUrl: 'http://edx.org' }} />
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('EnrollmentCodeQuantityUpdated', () => {
  it('should render with values', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore()}
        >
          <EnrollmentCodeQuantityUpdated values={{ quantity: 2, price: 100 }} />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TransactionDeclined', () => {
  it('should render with values', () => {
    const component = (
      <IntlProvider locale="en">
        <TransactionDeclined />
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
