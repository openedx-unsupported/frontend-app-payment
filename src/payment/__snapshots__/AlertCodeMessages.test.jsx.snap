// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollmentCodeQuantityUpdated should render with values 1`] = `
Array [
  <h6>
    We've updated your quantity.
  </h6>,
  <span>
    Your cart includes 2 enrollment codes at a total cost of 
    <span>
      $100.00
    </span>
    , that you will receive via email.
  </span>,
]
`;

exports[`SingleEnrollmentCodeWarning should render with values 1`] = `
Array [
  <h6>
    Purchasing just for yourself?
  </h6>,
  <span>
    If you are purchasing a single code for someone else, please continue with checkout. However, if you are the learner 
    <a
      className="pgn__hyperlink default-link standalone-link"
      href="http://edx.org"
      onClick={[Function]}
      target="_self"
    >
      <span>
        click here to enroll directly
      </span>
    </a>
    .
  </span>,
]
`;

exports[`TransactionDeclined should render with values 1`] = `
<span>
  Your payment could not be processed. Please check your payment information or reach out to your bank or financial institution for further assistance.
</span>
`;
